sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala

```diff
@@ -179,7 +179,7 @@ private[ui] abstract class ExecutionTable(
   }
 
   private def descriptionCell(execution: SQLExecutionUIData): Seq[Node] = {
-    val details = if (execution.details.nonEmpty) {
+    val details = if (execution.details != null) {
```

Comment:
Still worth checking for nonEmpty? the behavior may be different in the second change here otherwise.

Reply:
Seems previously detail link won't be shown for empty details, this change makes it shown anyway.

Reply:
+1, we can write `Option(execution.details).exist(_.nonEmpty)`, or simply `execution.details != null && execution.details.nonEmpty`

Reply:
How about make the default value of `details` an empty string? Then we don't have to worry about the `null` case here.

Reply:
Sure. That could further change the semantics somewhere else, though maybe not in any meaningful way. I suppose it means you can't distinguish empty and unset then. Overall I'd just add the null check here and leave it at that, but don't feel strongly about it.

Reply:
> How about make the default value of details an empty string?

I'd double check how that works with serialization before trying that. Jackson used to have issues with default argument values in case classes, but maybe that's been fixed.

The code suggested by Wenchen is safer, though, and avoids having to do that.